src/js/models/layerstructure.js

```diff
@@ -1317,6 +1318,37 @@ define(function (require, exports, module) {
     };
 
     /**
+     * Add a new stroke, described by a Photoshop descriptor, to the given layers.
+     * If strokeStyleDescriptor is a single object, it will be applied to all layers
+     * otherwise it should be a List of descriptors which corresponds by index to the provided layerIDs
+     *
+     * @param {Immutable.Iterable.<number>} layerIDs
+     * @param {object | Immutable.Iterable.<object>} strokeStyleDescriptor
+     * @return {LayerStructure}
+     */
+    LayerStructure.prototype.addStroke = function (layerIDs, strokeStyleDescriptor) {
+        var isList = Immutable.List.isList(strokeStyleDescriptor);
+       
+        var getStroke = function (index) {
+            return isList ?
+                Stroke.fromStrokeStyleDescriptor(strokeStyleDescriptor.get(index)) :
+                Stroke.fromStrokeStyleDescriptor(strokeStyleDescriptor);
+        };
+       
+        var nextLayers = Immutable.Map(layerIDs.reduce(function (map, layerID, index) {
```

Comment:
side note: I now believe (with very little actual evidence to back it up) that it is more efficient to map things to an array of two-element arrays, then create an Immutable.Map (which accepts that as a constructor format), and then merge that in to the main Map.  See here: https://github.com/adobe-photoshop/spaces-design/blob/master/src/js/models/documentexports.js#L261    food for thought anyway.